### ISSUE TITLE
6.4 -> 7.0.10 Failure

### ISSUE THREAD

**Issue by astrugatch** [grokability/snipe-it#15226]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

Getting the following error when upgrading from 6.4 to 7.0.10

```
--
   Error

  Call to undefined method Illuminate\View\Compilers\BladeCompiler::anonymousComponentPath()

  at vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php:261
    257▕         if (! $instance) {
    258▕             throw new RuntimeException('A facade root has not been set.');
    259▕         }
    260▕
  ➜ 261▕         return $instance->$method(...$args);
    262▕     }
    263▕ }
    264▕

  1   app/Providers/BladeServiceProvider.php:23
      Illuminate\Support\Facades\Facade::__callStatic()

      +7 vendor frames
  9   [internal]:0
      Illuminate\Foundation\Application::Illuminate\Foundation\{closure}()
```

I have been successfully using `php upgrade.php` for several versions now until this upgrade.

I DID get the update to run after a composer self update, but I reverted to an old snapshot because composer self update wouldn't work properly unless run as sudo. If I don't run it as sudo I get:

```
  [Composer\Downloader\FilesystemException]
  Filesystem exception:
  Composer update failed: "/usr/local/bin/composer" could not be written.
  rename(/home/snipeuser/.cache/composer/composer-temp.phar,/usr/local/bin/composer): Permission denied
```

### Reproduction steps

1. Upgrade from 6.4 to latest using php upgrade.php
2.
3.
...

### Expected behavior

Successful update without error.

### Snipe-IT Version

6.4

### Operating System

Ubuntu 20.04

### Web Server

Apache

### PHP Version

8.1.29

### Operating System

MacOS

### Browser

Chrome

### Version

latest

### Device

N/A

### Operating System

N/A

### Browser

N/A

### Version

N/A

**Comment by madridu** [comment]
I got the same issue trying to update to 7.0.10 using `php upgrade.php`

**Comment by snipe** [comment]
> because composer self update wouldn't work properly unless run as sudo. If I don't run it as sudo I get:

Can you run it as sudo then?

**Comment by snipe** [comment]
@madridu try running `composer self-update` to bring your composer up to the latest version

**Comment by madridu** [comment]
> @madridu try running `composer self-update` to bring your composer up to the latest version

Tried but not working. I had to uninstall composer and install it again to the latest version. Then update all packages.

After updating composer I was able to run update without failure.

**Comment by astrugatch** [comment]
@snipe 

Running composer self-update worked to resolve the issue but only when run as sudo. Just want to make sure this doesn’t fall under the “never run composer as root/sudo”

**Comment by snipe** [comment]
Yeah, you’re not supposed to run composer install as sudo, but depending on how you set it up, updating composer itself is usually ok to run as sudo

**Comment by astrugatch** [comment]
My composer is installed globally, so I’m assuming this is why it was necessary.

**Comment by curtishall** [comment]
I have the same problem, 6.3.3 -> 7 upgrade.  Composer is installed globally as well:

```
--------------------------------------------------------
STEP 4: Backing up database: 
--------------------------------------------------------

-- 
   Error

  Call to undefined method Illuminate\View\Compilers\BladeCompiler::anonymousComponentPath()

  at vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php:261
    257▕         if (! $instance) {
    258▕             throw new RuntimeException('A facade root has not been set.');
    259▕         }
    260▕
  ➜ 261▕         return $instance->$method(...$args);
    262▕     }
    263▕ }
    264▕

  1   app/Providers/BladeServiceProvider.php:23
      Illuminate\Support\Facades\Facade::__callStatic()

      +7 vendor frames 
  9   [internal]:0
      Illuminate\Foundation\Application::Illuminate\Foundation\{closure}()

Something went wrong with your backup. Aborting!

```


````
www-data@snipeit-2:~/snipe-it$ composer self-update
You are already using the latest available Composer version 2.7.8 (stable channel).
```

**Comment by billtzim** [comment]
> I have the same problem, 6.3.3 -> 7 upgrade. Composer is installed globally as well:
> 
> ```
> --------------------------------------------------------
> STEP 4: Backing up database: 
> --------------------------------------------------------
> 
> -- 
>    Error
> 
>   Call to undefined method Illuminate\View\Compilers\BladeCompiler::anonymousComponentPath()
> 
>   at vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php:261
>     257▕         if (! $instance) {
>     258▕             throw new RuntimeException('A facade root has not been set.');
>     259▕         }
>     260▕
>   ➜ 261▕         return $instance->$method(...$args);
>     262▕     }
>     263▕ }
>     264▕
> 
>   1   app/Providers/BladeServiceProvider.php:23
>       Illuminate\Support\Facades\Facade::__callStatic()
> 
>       +7 vendor frames 
>   9   [internal]:0
>       Illuminate\Foundation\Application::Illuminate\Foundation\{closure}()
> 
> Something went wrong with your backup. Aborting!
> ```
> 
> ```
> www-data@snipeit-2:~/snipe-it$ composer self-update
> You are already using the latest available Composer version 2.7.8 (stable channel).
> ```
> ```

Same thing happened to me, exactly same versions. Also already in latest composer

**UPDATE**: after following the instruction and commands [here](https://snipe-it.readme.io/docs/upgrading#after-pulling-the-new-version-files) problem was solved
